Documentation/Administration/Production/ReverseProxy/Index.rst

```diff
@@ -0,0 +1,159 @@
+:navigation-title: Reverse Proxy
+
+..  include:: /Includes.rst.txt
+.. _reverse-proxy-setup:
+
+====================================
+Running TYPO3 behind a reverse proxy
+====================================
+
+Many TYPO3 production environments use a reverse proxy such as NGINX,
+Apache (with mod_proxy), Traefik, or a cloud load balancer. This is especially
+common in container-based deployments using Docker or Kubernetes, where a
+reverse proxy helps manage access to multiple services. Reverse proxies
+typically handle HTTPS termination, request routing, and security features
+such as rate limiting or header filtering.
+
+To ensure TYPO3 works correctly in this setup, some additional configuration is
+required. TYPO3 must be told to trust the proxy headers to interpret the
+correct request scheme and client IP.
+
+..  contents:: Table of contents
+
+..  _reverse-proxy-typo3-configuration:
+
+Configuring TYPO3 to trust a reverse proxy
+==========================================
+
+TYPO3 needs to be explicitly configured to accept reverse proxy headers and to
+trust specific proxy addresses.
+
+Add the following lines to :file:`config/system/additional.php`:
+
+..  code-block:: php
+    :caption: config/system/additional.php
+
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['reverseProxyIP'] = '192.0.2.1,198.51.100.5';
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['reverseProxySSL'] = '192.0.2.1,198.51.100.5';
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['reverseProxyHeaderMultiValue'] = 'first';
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['trustedHostsPattern'] = '^(www\.)?example\.com$';
+
+..  seealso::
+
+    The following settings in :php:`$GLOBALS['TYPO3_CONF_VARS']['SYS']`:
+
+    *   `reverseProxyIP  <https://docs.typo3.org/permalink/t3coreapi:confval-globals-typo3-conf-vars-sys-reverseproxyip>`_
+    *   `reverseProxyHeaderMultiValue  <https://docs.typo3.org/permalink/t3coreapi:confval-globals-typo3-conf-vars-sys-reverseproxyheadermultivalue>`_
+    *   `reverseProxyPrefix  <https://docs.typo3.org/permalink/t3coreapi:confval-globals-typo3-conf-vars-sys-reverseproxyprefix>`_
+    *   `reverseProxyPrefixSSL  <https://docs.typo3.org/permalink/t3coreapi:confval-globals-typo3-conf-vars-sys-reverseproxyprefixssl>`_
+    *   `trustedHostsPattern  <https://docs.typo3.org/permalink/t3coreapi:confval-globals-typo3-conf-vars-sys-trustedhostspattern>`_
+
+In production, always use specific, individual IP addresses instead of
+wildcards. Each IP must be listed in a comma-separated string.
+
+..  note::
+
+    In development or internal testing environments, wildcard values such as
+    `'*'` may be used temporarily for convenience. However, they must **never**
+    be used in public or production-facing deployments, as they allow untrusted
+    sources to spoof critical headers such as `X-Forwarded-For` or
+    `X-Forwarded-Proto`. This can lead to incorrect URL generation,
+    security misconfigurations, and exposure of backend systems.
```

Comment:
What are the backend systems you mention here? I think the url and security implications should be mentioned. But backend systems seem to be out of context?

Reply:
I removed this part and only kept the information I am more sure about. There was a securtiy bulletin where invalid settings made the backend accessible by stealing sessions somehow but I think that is resolved in all secure TYPO3 versions.